src/protocol/sort/shuffle.rs

```diff
@@ -149,6 +149,15 @@ impl<'a, F: Field, S: Step + SpaceIndex> Shuffle<'a, F, S> {
         *self.input = self.single_shuffle(ctx, Step2).await.unwrap();
         *self.input = self.single_shuffle(ctx, Step3).await.unwrap();
 
+        assert_ne!(
+            &ctx.participant[(self.step_fn)(Step1)],
+            &ctx.participant[(self.step_fn)(Step2)]
```

Comment:
I'm not super familiar with this bit of the infra. Is the "participant" just a PRSS concept?

Reply:
yes it is, participant is vector of prss (i.e. for all the steps) whereas prss is for a single step. I think @akoshelev is looking at changing the name of participant to something more obvious.